starlite/openapi/controller.py

```diff
@@ -65,6 +67,57 @@ def get_schema_from_request(request: Request) -> "OpenAPI":
             raise ImproperlyConfiguredException("Starlite has not been instantiated with OpenAPIConfig")
         return request.app.openapi_schema
 
+    @staticmethod
+    def check_exposed_endpoints(request: Request) -> Optional[Response]:
```

Comment:
Sorry if it got a bit confusing in the issue discussion before, but we came to the conclusion that this needed to be a method that was called from within the route handlers to return the response, rather than via the `before_request` hook, so that the method could be easily overridden if desired.

This was also the reason that I suggested we change the response type of the handler methods so that they return a `Response` object in all cases, otherwise it would have been fine to leave those as `str` if we were going to use the `before_request` approach.

Pseudo code, but something along these lines:

```python
    @get(path="/elements", include_in_schema=False)
    def stoplight_elements(self, request: Request) -> Response:
        if should_serve_elements():
            # return the 200 response
        else:
            # return the 404 response
```
It'll work around those validation errors in CI too

Reply:
Alright, thanks for the clarification.
I'll change it o7